File: mycollab-mobile/src/main/java/com/esofthead/mycollab/mobile/module/project/view/parameters/TaskScreenData.java
package com.esofthead.mycollab.mobile.module.project.view.parameters;

import com.esofthead.mycollab.module.crm.domain.SimpleTask;
import com.esofthead.mycollab.vaadin.mvp.ScreenData;

/**
 * @author MyCollab Ltd.
 *
 * @since 4.5.0
 *
 */
public class TaskScreenData {
	public static class List extends ScreenData<Integer> {
		public List(Integer param) {
			super(param);
		}
	}

	public static class Add extends ScreenData<Integer> {
		public Add() {
			super(null);
		}
	}

	public static class Read extends ScreenData<Integer> {
		public Read(Integer param) {
			super(param);
		}
	}

	public static class Edit extends ScreenData<SimpleTask> {
		public Edit(SimpleTask param) {
			super(param);
		}
	}
}

File: mycollab-mobile/src/main/scala/com/esofthead/mycollab/mobile/module/project/view/parameters/TaskScreenData.scala
package com.esofthead.mycollab.mobile.module.project.view.parameters

import com.esofthead.mycollab.module.project.domain.SimpleTask
import com.esofthead.mycollab.module.project.domain.criteria.TaskSearchCriteria
import com.esofthead.mycollab.vaadin.mvp.ScreenData

/**
  * @author MyCollab Ltd
  * @since 5.2.5
  */
object TaskScreenData {

  class Search(param: TaskSearchCriteria) extends ScreenData(param) {}

  class Read(param: Integer) extends ScreenData(param) {}

  class Edit(param: SimpleTask) extends ScreenData(param) {}

  class Add(param: SimpleTask) extends ScreenData(param) {}

}
